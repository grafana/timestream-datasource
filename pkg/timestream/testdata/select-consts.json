{
    "ColumnInfo": [
        {
            "Name": "t_int32",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "INTEGER",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "t_varchar",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "VARCHAR",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "timestamp",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "TIMESTAMP",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "interval_day_to_second",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "INTERVAL_DAY_TO_SECOND",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "interval_year_to_month",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "INTERVAL_YEAR_TO_MONTH",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "time",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "TIME",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        },
        {
            "Name": "date",
            "Type": {
                "ArrayColumnInfo": null,
                "RowColumnInfo": null,
                "ScalarType": "DATE",
                "TimeSeriesMeasureValueColumnInfo": null
            }
        }
    ],
    "NextToken": null,
    "QueryId": "#QueryId#",
    "QueryStatus": {
        "CumulativeBytesMetered": 10000000,
        "CumulativeBytesScanned": 0,
        "ProgressPercentage": 100
    },
    "Rows": [
        {
            "Data": [
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "1",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "two",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2020-08-08 01:00:00.000000000",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "1 21:00:00.000000000",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2-7",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "01:00:00.000000000",
                    "TimeSeriesValue": null
                },
                {
                    "ArrayValue": null,
                    "NullValue": null,
                    "RowValue": null,
                    "ScalarValue": "2020-08-08",
                    "TimeSeriesValue": null
                }
            ]
        }
    ]
}