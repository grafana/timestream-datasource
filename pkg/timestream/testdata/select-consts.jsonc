//  ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ
//  
//  Frame[0] {
//      "typeVersion": [
//          0,
//          0
//      ]
//  }
//  Name: 
//  Dimensions: 7 Fields by 1 Rows
//  +----------------+-----------------+-------------------------------+------------------------------+------------------------------+-------------------------------+-------------------------------+
//  | Name: t_int32  | Name: t_varchar | Name: timestamp               | Name: interval_day_to_second | Name: interval_year_to_month | Name: time                    | Name: date                    |
//  | Labels:        | Labels:         | Labels:                       | Labels:                      | Labels:                      | Labels:                       | Labels:                       |
//  | Type: []*int32 | Type: []*string | Type: []*time.Time            | Type: []*string              | Type: []*string              | Type: []*time.Time            | Type: []*time.Time            |
//  +----------------+-----------------+-------------------------------+------------------------------+------------------------------+-------------------------------+-------------------------------+
//  | 1              | two             | 2020-08-08 01:00:00 +0000 UTC | 1 21:00:00.000000000         | 2-7                          | 1970-01-01 01:00:00 +0000 UTC | 2020-08-08 00:00:00 +0000 UTC |
//  +----------------+-----------------+-------------------------------+------------------------------+------------------------------+-------------------------------+-------------------------------+
//  
//  
//  ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ
{
  "status": 200,
  "frames": [
    {
      "schema": {
        "meta": {
          "typeVersion": [
            0,
            0
          ]
        },
        "fields": [
          {
            "name": "t_int32",
            "type": "number",
            "typeInfo": {
              "frame": "int32",
              "nullable": true
            }
          },
          {
            "name": "t_varchar",
            "type": "string",
            "typeInfo": {
              "frame": "string",
              "nullable": true
            }
          },
          {
            "name": "timestamp",
            "type": "time",
            "typeInfo": {
              "frame": "time.Time",
              "nullable": true
            }
          },
          {
            "name": "interval_day_to_second",
            "type": "string",
            "typeInfo": {
              "frame": "string",
              "nullable": true
            }
          },
          {
            "name": "interval_year_to_month",
            "type": "string",
            "typeInfo": {
              "frame": "string",
              "nullable": true
            }
          },
          {
            "name": "time",
            "type": "time",
            "typeInfo": {
              "frame": "time.Time",
              "nullable": true
            }
          },
          {
            "name": "date",
            "type": "time",
            "typeInfo": {
              "frame": "time.Time",
              "nullable": true
            }
          }
        ]
      },
      "data": {
        "values": [
          [
            1
          ],
          [
            "two"
          ],
          [
            1596848400000
          ],
          [
            "1 21:00:00.000000000"
          ],
          [
            "2-7"
          ],
          [
            3600000
          ],
          [
            1596844800000
          ]
        ]
      }
    }
  ]
}