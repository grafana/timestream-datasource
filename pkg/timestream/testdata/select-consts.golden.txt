ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ

Frame[0] {}
Name: 
Dimensions: 7 Fields by 1 Rows
+----------------+-----------------+-------------------------------+------------------------------+------------------------------+-------------------------------+-------------------------------+
| Name: t_int32  | Name: t_varchar | Name: timestamp               | Name: interval_day_to_second | Name: interval_year_to_month | Name: time                    | Name: date                    |
| Labels:        | Labels:         | Labels:                       | Labels:                      | Labels:                      | Labels:                       | Labels:                       |
| Type: []*int32 | Type: []*string | Type: []time.Time             | Type: []*string              | Type: []*string              | Type: []*time.Time            | Type: []*time.Time            |
+----------------+-----------------+-------------------------------+------------------------------+------------------------------+-------------------------------+-------------------------------+
| 1              | two             | 2020-08-08 01:00:00 +0000 UTC | 1 21:00:00.000000000         | 2-7                          | 1970-01-01 01:00:00 +0000 UTC | 2020-08-08 00:00:00 +0000 UTC |
+----------------+-----------------+-------------------------------+------------------------------+------------------------------+-------------------------------+-------------------------------+


====== TEST DATA RESPONSE (arrow base64) ======
FRAME=QVJST1cxAAD/////0AMAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEDAAoADAAAAAgABAAKAAAACAAAAHQAAAADAAAATAAAACgAAAAEAAAAxPz//wgAAAAMAAAAAAAAAAAAAAAFAAAAcmVmSWQAAADk/P//CAAAAAwAAAAAAAAAAAAAAAQAAABuYW1lAAAAAAT9//8IAAAADAAAAAIAAAB7fQAABAAAAG1ldGEAAAAABwAAALgCAABAAgAA0AEAAEABAADEAAAAZAAAAAQAAAB2/f//FAAAADwAAAA8AAAAAAAKATwAAAABAAAABAAAAGT9//8IAAAAEAAAAAQAAABkYXRlAAAAAAQAAABuYW1lAAAAAAAAAAA+/v//AAADAAQAAABkYXRlAAAAANL9//8UAAAAPAAAADwAAAAAAAoBPAAAAAEAAAAEAAAAwP3//wgAAAAQAAAABAAAAHRpbWUAAAAABAAAAG5hbWUAAAAAAAAAAJr+//8AAAMABAAAAHRpbWUAAAAALv7//xQAAABMAAAATAAAAAAABQFIAAAAAQAAAAQAAAAc/v//CAAAACAAAAAWAAAAaW50ZXJ2YWxfeWVhcl90b19tb250aAAABAAAAG5hbWUAAAAAAAAAAJz+//8WAAAAaW50ZXJ2YWxfeWVhcl90b19tb250aAAApv7//xQAAABMAAAATAAAAAAABQFIAAAAAQAAAAQAAACU/v//CAAAACAAAAAWAAAAaW50ZXJ2YWxfZGF5X3RvX3NlY29uZAAABAAAAG5hbWUAAAAAAAAAABT///8WAAAAaW50ZXJ2YWxfZGF5X3RvX3NlY29uZAAAAAASABgAFAAAABMADAAAAAgABAASAAAAFAAAAEAAAABIAAAAAAAACkgAAAABAAAABAAAACD///8IAAAAFAAAAAkAAAB0aW1lc3RhbXAAAAAEAAAAbmFtZQAAAAAAAAAAAAAGAAgABgAGAAAAAAADAAkAAAB0aW1lc3RhbXAAAACe////FAAAAEAAAABEAAAAAAAFAUAAAAABAAAABAAAAIz///8IAAAAFAAAAAkAAAB0X3ZhcmNoYXIAAAAEAAAAbmFtZQAAAAAAAAAABAAEAAQAAAAJAAAAdF92YXJjaGFyABIAGAAUABMAEgAMAAAACAAEABIAAAAUAAAARAAAAEwAAAAAAAIBUAAAAAEAAAAMAAAACAAMAAgABAAIAAAACAAAABAAAAAHAAAAdF9pbnQzMgAEAAAAbmFtZQAAAAAAAAAACAAMAAgABwAIAAAAAAAAASAAAAAHAAAAdF9pbnQzMgD/////2AEAABQAAAAAAAAADAAWABQAEwAMAAQADAAAAGAAAAAAAAAAFAAAAAAAAAMDAAoAGAAMAAgABAAKAAAAFAAAACgBAAABAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAIAAAAAAAAABAAAAAAAAAACAAAAAAAAAAYAAAAAAAAAAAAAAAAAAAAGAAAAAAAAAAIAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAgAAAAAAAAAKAAAAAAAAAAYAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAgAAAAAAAAASAAAAAAAAAAIAAAAAAAAAFAAAAAAAAAAAAAAAAAAAABQAAAAAAAAAAgAAAAAAAAAWAAAAAAAAAAAAAAAAAAAAFgAAAAAAAAACAAAAAAAAAAAAAAABwAAAAEAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAADAAAAdHdvAAAAAAAAoCbKKCUpFgAAAAAUAAAAMSAyMTowMDowMC4wMDAwMDAwMDAAAAAAAAAAAAMAAAAyLTcAAAAAAACguDBGAwAAAABumeIhKRYQAAAADAAUABIADAAIAAQADAAAABAAAAAsAAAAPAAAAAAAAwABAAAA4AMAAAAAAADgAQAAAAAAAGAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKAAwAAAAIAAQACgAAAAgAAAB0AAAAAwAAAEwAAAAoAAAABAAAAMT8//8IAAAADAAAAAAAAAAAAAAABQAAAHJlZklkAAAA5Pz//wgAAAAMAAAAAAAAAAAAAAAEAAAAbmFtZQAAAAAE/f//CAAAAAwAAAACAAAAe30AAAQAAABtZXRhAAAAAAcAAAC4AgAAQAIAANABAABAAQAAxAAAAGQAAAAEAAAAdv3//xQAAAA8AAAAPAAAAAAACgE8AAAAAQAAAAQAAABk/f//CAAAABAAAAAEAAAAZGF0ZQAAAAAEAAAAbmFtZQAAAAAAAAAAPv7//wAAAwAEAAAAZGF0ZQAAAADS/f//FAAAADwAAAA8AAAAAAAKATwAAAABAAAABAAAAMD9//8IAAAAEAAAAAQAAAB0aW1lAAAAAAQAAABuYW1lAAAAAAAAAACa/v//AAADAAQAAAB0aW1lAAAAAC7+//8UAAAATAAAAEwAAAAAAAUBSAAAAAEAAAAEAAAAHP7//wgAAAAgAAAAFgAAAGludGVydmFsX3llYXJfdG9fbW9udGgAAAQAAABuYW1lAAAAAAAAAACc/v//FgAAAGludGVydmFsX3llYXJfdG9fbW9udGgAAKb+//8UAAAATAAAAEwAAAAAAAUBSAAAAAEAAAAEAAAAlP7//wgAAAAgAAAAFgAAAGludGVydmFsX2RheV90b19zZWNvbmQAAAQAAABuYW1lAAAAAAAAAAAU////FgAAAGludGVydmFsX2RheV90b19zZWNvbmQAAAAAEgAYABQAAAATAAwAAAAIAAQAEgAAABQAAABAAAAASAAAAAAAAApIAAAAAQAAAAQAAAAg////CAAAABQAAAAJAAAAdGltZXN0YW1wAAAABAAAAG5hbWUAAAAAAAAAAAAABgAIAAYABgAAAAAAAwAJAAAAdGltZXN0YW1wAAAAnv///xQAAABAAAAARAAAAAAABQFAAAAAAQAAAAQAAACM////CAAAABQAAAAJAAAAdF92YXJjaGFyAAAABAAAAG5hbWUAAAAAAAAAAAQABAAEAAAACQAAAHRfdmFyY2hhcgASABgAFAATABIADAAAAAgABAASAAAAFAAAAEQAAABMAAAAAAACAVAAAAABAAAADAAAAAgADAAIAAQACAAAAAgAAAAQAAAABwAAAHRfaW50MzIABAAAAG5hbWUAAAAAAAAAAAgADAAIAAcACAAAAAAAAAEgAAAABwAAAHRfaW50MzIAAAQAAEFSUk9XMQ==
